### PR TITLE
Updated the linearWeight function to use the new 'nominal' function in JointTrajectoryPt

### PR DESCRIPTION
**Pull request by Jmeyer1292**
This reduces copying by a huge amount, allowing the graph to build twice as fast for edge heavy graphs. The diff below looks kinda gnarly but all I did was remove the outer-most if check and the cruft related to calling `getJointPoses`.

### CODE REVIEW

File: descartes_planner/src/planning_graph.cpp [hunk 1]
```diff
@@ -996,48 +996,33 @@ PlanningGraph::LinearWeightResult PlanningGraph::linearWeight(const JointTraject
...
-      double vector_diff = 0;
-      for (unsigned i = 0; i < start_vector.size(); i++)
-      {
-        double joint_diff = std::abs(end_vector[i] - start_vector[i]);
-        vector_diff += joint_diff ;
-      }
-
-      result.first = true;
-      result.second = vector_diff;
       return result;
```
Comment (jrgnicho) [inline descartes_planner/src/planning_graph.cpp:1008]: Nothing is being assigned to the **result** object any more, is that the intended behaviour?

File: descartes_planner/src/planning_graph.cpp [hunk 2]
```diff
@@ -996,48 +996,33 @@ PlanningGraph::LinearWeightResult PlanningGraph::linearWeight(const JointTraject
...
-    const std::vector<double>& end_vector = joint_poses_end[0];
-    if (start_vector.size() == end_vector.size())
+  const std::vector<double>& start_vector = start.nominal();
+  const std::vector<double>& end_vector = end.nominal();
+  if (start_vector.size() == end_vector.size())
+  {
+    // Check to see if time is specified and if so, check to see if the
+    // joint motion is possible in the window provided
+    if (end.getTiming().isSpecified() &&
+        !robot_model_->isValidMove(start_vector, end_vector, end.getTiming().upper))
```
Comment (jrgnicho) [inline descartes_planner/src/planning_graph.cpp:1006]: This check feels non-intuitive to me.   Why is the returned bool from isValidMove() being negated?

File: descartes_planner/src/planning_graph.cpp (hunk 1 above)
Comment (Jmeyer1292) [inline descartes_planner/src/planning_graph.cpp:1008]: result.first is set to false at the top of the function. If you look at the annotation-free version you'll see it still makes sense.

Comment (jrgnicho) [inline descartes_planner/src/planning_graph.cpp:1008]: +1

File: descartes_planner/src/planning_graph.cpp (hunk 2 above)
Comment (Jmeyer1292) [inline descartes_planner/src/planning_graph.cpp:1006]: If the timing is specified and the move is NOT valid, then we return a false immediately.

Comment (jrgnicho) [inline descartes_planner/src/planning_graph.cpp:1006]: +1

**Comment by jrgnicho** [comment]
Did you by any chance validate these changes by running them on a simulated robot?

**Comment by Jmeyer1292** [comment]
I did. I tested it on my SIA20D setup for Descartes.

**Comment by jrgnicho** [comment]
Great.  +1 from me, feel free to merge

**Pull request merged**